internal/core/metadata/operators/device_service/get.go

```diff
@@ -53,29 +53,52 @@ func (op deviceServiceLoadAll) Execute() (services []contract.DeviceService, err
 	return
 }
 
-type deviceServiceLoadByAddressableId struct {
-	id string
-	db DeviceServiceLoader
+type deviceServiceLoadByAddressable struct {
+	id   string
+	name string
+	db   DeviceServiceLoader
+}
+
+// NewDeviceServiceLoadByAddressableName creates a new Executor that retrieves all DeviceService associated with a given Addressable name.
+func NewDeviceServiceLoadByAddressableName(name string, db DeviceServiceLoader) DeviceServiceGetAllExecutor {
```

Comment:
If the `DeviceServiceGetAllExecutor` interface is going to be multi-purposed due to its conformant signature, would recommend renaming it since we aren't "getting all" here. Perhaps something like `DeviceServiceListExecutor`.